### PR TITLE
[T3A3][W14-A4]Tan Wei

### PR DESCRIPTION
**Pull request by tanwei0319**

### CODE REVIEW

**Comment by tanwei0319** [comment]
Ready for review

**Comment by tanwei0319** [comment]
Ready for review

File: test/java/seedu/addressbook/parser/ParserTest.java [hunk 1]
```diff
@@ -271,7 +271,7 @@ private static String convertPersonToAddCommandString(ReadOnlyPerson person) {
                 + person.getName().fullName
                 + (person.getPhone().isPrivate() ? " pp/" : " p/") + person.getPhone().value
                 + (person.getEmail().isPrivate() ? " pe/" : " e/") + person.getEmail().value
-                + (person.getAddress().isPrivate() ? " pa/" : " a/") + person.getAddress().value;
+                + (person.getAddress().isPrivate() ? " pa/" : " a/") + person.getAddress().toString();
         for (Tag tag : person.getTags()) {
```
Comment (K1ang) [inline test/java/seedu/addressbook/parser/ParserTest.java:275]: Good to see the relevant tests are being updated!

File: src/seedu/addressbook/data/person/Block.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package seedu.addressbook.data.person;
+
```
Comment (K1ang) [inline src/seedu/addressbook/data/person/Block.java:2]: Remember to include header comments for the classes!

File: src/seedu/addressbook/data/person/Address.java [hunk 3]
```diff
@@ -1,59 +1,109 @@
...
+        return fullAddress;
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+                && this.toString().equals(((Address) other).toString())); // state check
     }
```
Comment (K1ang) [inline src/seedu/addressbook/data/person/Address.java:98]: Using each smaller classes for comparison may be a better option.

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -1,59 +1,109 @@
...
+    	}
+    	
+    	// Postal code is not 6-digit integer
+    	int postalCodeInt = Integer.parseInt(trimmedAddress[ADDRESS_POSTALCODE_INDEX].trim());
+    	if (postalCodeInt < 100000 || postalCodeInt > 999999) {
+    		return false;
+    	}
+    	
+    	return true;
     }
```
Comment (K1ang) [inline src/seedu/addressbook/data/person/Address.java:83]: How about using each classes for their own specific validation?

**Review by K1ang — COMMENTED** [review]
@tanwei0319 Some comments added. Please close the PR after reading comments.
Keep up the good work! Take note that relevant documentations should be updated too.

**Pull request closed without merging**